aws/resource_aws_db_proxy_endpoint.go

```diff
@@ -0,0 +1,229 @@
+	dbProxyEndpointName := d.Get("db_proxy_endpoint_name").(string)
+
+	params := rds.CreateDBProxyEndpointInput{
+		DBProxyName:         aws.String(dbProxyName),
+		DBProxyEndpointName: aws.String(dbProxyEndpointName),
+		TargetRole:          aws.String(d.Get("target_role").(string)),
+		VpcSubnetIds:        expandStringSet(d.Get("vpc_subnet_ids").(*schema.Set)),
+		Tags:                keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().RdsTags(),
+	}
+
+	if v := d.Get("vpc_security_group_ids").(*schema.Set); v.Len() > 0 {
+		params.VpcSecurityGroupIds = expandStringSet(v)
+	}
+
+	_, err := conn.CreateDBProxyEndpoint(&params)
+
+	if err != nil {
+		return fmt.Errorf("error Creating RDS DB Proxy Endpoint (%s/%s): %w", dbProxyName, dbProxyEndpointName, err)
+	}
+
+	d.SetId(strings.Join([]string{dbProxyName, dbProxyEndpointName}, "/"))
+
+	if _, err := waiter.DBProxyEndpointAvailable(conn, d.Id(), d.Timeout(schema.TimeoutCreate)); err != nil {
+		return fmt.Errorf("error waiting for RDS DB Proxy Endpoint (%s) to become available: %w", d.Id(), err)
+	}
+
+	return resourceAwsDbProxyEndpointRead(d, meta)
+}
+
+func resourceAwsDbProxyEndpointRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	dbProxyEndpoint, err := finder.DBProxyEndpoint(conn, d.Id())
+
+	if tfawserr.ErrCodeEquals(err, rds.ErrCodeDBProxyNotFoundFault) {
+		log.Printf("[WARN] RDS DB Proxy Endpoint (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if tfawserr.ErrCodeEquals(err, rds.ErrCodeDBProxyEndpointNotFoundFault) {
+		log.Printf("[WARN] RDS DB Proxy Endpoint (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error reading RDS DB Proxy Endpoint (%s): %w", d.Id(), err)
+	}
+
+	if dbProxyEndpoint == nil {
+		log.Printf("[WARN] RDS DB Proxy Endpoint (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	endpointArn := aws.StringValue(dbProxyEndpoint.DBProxyEndpointArn)
+	d.Set("arn", endpointArn)
+	d.Set("db_proxy_name", dbProxyEndpoint.DBProxyName)
+	d.Set("endpoint", dbProxyEndpoint.Endpoint)
+	d.Set("db_proxy_endpoint_name", dbProxyEndpoint.DBProxyEndpointName)
+	d.Set("is_default", dbProxyEndpoint.IsDefault)
+	d.Set("target_role", dbProxyEndpoint.TargetRole)
+	d.Set("vpc_id", dbProxyEndpoint.VpcId)
+	d.Set("target_role", dbProxyEndpoint.TargetRole)
+	d.Set("vpc_subnet_ids", flattenStringSet(dbProxyEndpoint.VpcSubnetIds))
+	d.Set("vpc_security_group_ids", flattenStringSet(dbProxyEndpoint.VpcSecurityGroupIds))
+
+	tags, err := keyvaluetags.RdsListTags(conn, endpointArn)
+
+	if err != nil {
+		return fmt.Errorf("Error listing tags for RDS DB Proxy Endpoint (%s): %w", endpointArn, err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("Error setting tags: %w", err)
+	}
```

Comment:
Default tags support 🎉 

```suggestion
	tags = tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig)

	//lintignore:AWSR002
	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
		return fmt.Errorf("error setting tags: %w", err)
	}

	if err := d.Set("tags_all", tags.Map()); err != nil {
		return fmt.Errorf("error setting tags_all: %w", err)
	}
```